test/common/devCompileOptions.cmake

```diff
@@ -157,7 +157,13 @@ ELSE()
         IF(alpaka_ACC_GPU_HIP_ENABLE)
             LIST(APPEND alpaka_DEV_COMPILE_OPTIONS "-Wno-unused-command-line-argument")
         ENDIF()
-    # ICC
+
+        # ICPX
+        if (${CMAKE_CXX_COMPILER_ID} STREQUAL "IntelLLVM")
+            # fast math is turned on by default, which breaks our unit tests
+            LIST(APPEND alpaka_DEV_COMPILE_OPTIONS "-fp-model=precise")
```

Comment:
This, on the other hand, might break. Before CMake 3.20 `icpx` is identified as `Clang`.

Reply:
I changed it to clang's `-ffp-model=precise` and add it for both, clang and icpx. Let's see what the CI says.

Reply:
I only added it for icpx now, see below.